gensim/similarities/levenshtein.py

```diff
@@ -39,6 +39,9 @@ def levsim(t1, t2, alpha=1.8, beta=5.0):
         The multiplicative factor alpha defined by Charlet and Damnati (2017).
     beta : float
         The exponential factor beta defined by Charlet and Damnati (2017).
+    max_distance : {int,float}
+        The maximum Levenshtein distance between t1, and t2. Larger distances
+        will result in the Levenshtein similarity of zero.
```

Comment:
Why zero? That sounds like a bad idea -- zero distance means the opposite = "terms identical", not "terms too far apart to care".

A better value is `max_distance` or `inf`.

Reply:
Ah, I see you're converting distance to <0.0, 1.0> similarity.

In that case 0.0 is OK (same as dist=inf), but the change of units is confusing. Why not have the user input `min_similarity` instead of `max_distance`? If similarity is the conceptual unit we chose (as opposed to "distance"), we should stick to it.

It looks like we can freely convert between similarity and distance (bijection), so this is a more a design/conceptual question than a technical one.

Reply:
In commit 5154569, I ended up splitting the original function into the `levdist` function (with the `max_distance` parameter), which serves as a wrapper for the `distance` function from `python-Levenshtein`, and the `levsim` function (with the `min_similarity` parameter), which computes pointwise Levenshtein similarity in the sense of Damnati et al., 2017.

Instead of returning `float("inf")` when the distance exceeds `max_distance`, the current implementation of `levdist` returns `max(len(t1), len(t2))`, which is the upper bound on the distance between terms `t1`, and `t2`. This is a little more difficult to check for on the side of the application, but has the upside of removing the return type polymorphism (we always get an `int`) and it nicely plugs into the `levsim` function, where `max(len(t1), len(t2))` produces the similarity of zero.